project-api/test-artifacts/compiler-plugin-with-analyzer-generator-modifier/src/main/java/io.gayal.plugins.combined/CombinedCompilerPlugin.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gayal.plugins.combined;
+
+import io.ballerina.projects.plugins.CodeAnalysisContext;
+import io.ballerina.projects.plugins.CodeAnalyzer;
+import io.ballerina.projects.plugins.CodeGenerator;
+import io.ballerina.projects.plugins.CodeGeneratorContext;
+import io.ballerina.projects.plugins.CodeModifier;
+import io.ballerina.projects.plugins.CodeModifierContext;
+import io.ballerina.projects.plugins.CompilerPlugin;
+import io.ballerina.projects.plugins.CompilerPluginContext;
+
+public class CombinedCompilerPlugin extends CompilerPlugin {
+    @Override
+    public void init(CompilerPluginContext pluginContext) {
+        pluginContext.addCodeAnalyzer(new CodeAnalyzer() {
+            @Override
+            public void init(CodeAnalysisContext codeAnalysisContext) {
+                System.out.println("Hello from the analyzer");
+            }
+        });
+        pluginContext.addCodeGenerator(new CodeGenerator() {
+            @Override
+            public void init(CodeGeneratorContext codeGeneratorContext) {
+                System.out.println("Hello from the generator");
```

Comment:
Shall we change this to report a diagnostic?

Reply:
`runCodeModifers()`, `runCodeGenerators()` methods ignore the diagnostics from the code analyzing part even when the analyzers are run due to the identified bug. Therefore we do not receive the diagnostics from the code analyzer at the code modifier execution stage anyway. 

Had to use an sout as a result.

Reply:
I think an sout fails the checkstyles checks. 
May be return the error and report the diagnostic at a different level?

Reply:
Added the logs to a file and checked